### PR TITLE
[BACKLOG-40336] Link some echart properties with Chart Options

### PR DESCRIPTION
**Pull request by AAravind27**
This PR  :
- addresses review comments of #1753 
- allows user to adjust following chart properties via chart options for echarts
  - background color, 
  - label's font, size, style, & color, 
  - position of legend, background color of legend, each legends label's font, size, style, color

### CODE REVIEW

**Comment by buildguy** [comment]
<div align='center'>

[👍 Frogbot scanned this pull request and found that it did not add vulnerable dependencies.](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>

<details>
<summary> <b>Note:</b> </summary>


---
<div align='center'>

**Frogbot** also supports **Contextual Analysis, Secret Detection, IaC and SAST Vulnerabilities Scanning**. This features are included as part of the [JFrog Advanced Security](https://jfrog.com/advanced-security) package, which isn't enabled on your system.

</div>


</details>


---
<div align='center'>

[🐸 JFrog Frogbot](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 10m 22s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -Djs.no.sandbox
```
---
:ok_hand: All tests passed!
 
Tests run: **5240**, Failures: **0**, Skipped: **5** &nbsp;&nbsp; [Test Results](https://wingman.orl.eng.hitachivantara.com/job/projects/job/common-ui/job/PR-1760/1/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
<div align='center'>

[👍 Frogbot scanned this pull request and found that it did not add vulnerable dependencies.](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>

<details>
<summary> <b>Note:</b> </summary>


---
<div align='center'>

**Frogbot** also supports **Contextual Analysis, Secret Detection, IaC and SAST Vulnerabilities Scanning**. This features are included as part of the [JFrog Advanced Security](https://jfrog.com/advanced-security) package, which isn't enabled on your system.

</div>


</details>


---
<div align='center'>

[🐸 JFrog Frogbot](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 9m 56s
Build command:
```console
mvn clean verify -B -e -Daudit -Djs.no.sandbox
```
---
:ok_hand: All tests passed!
 
Tests run: **5240**, Failures: **0**, Skipped: **5** &nbsp;&nbsp; [Test Results](https://wingman.orl.eng.hitachivantara.com/job/projects/job/common-ui/job/PR-1760/2/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

File: impl/client/src/main/javascript/web/pentaho/visual/views/echarts/Abstract.js [hunk 1]
```diff
@@ -71,48 +73,105 @@ define([
...
+            x2: 0, y2: 1,
+            colorStops: [
+              {offset: 0, color: bgColorEnd},
+              {offset: 1, color: bgColorFill}
+            ],
+          };
+        }
+      } else {
+        return "transparent";
+      }
```
Comment (dcleao) [inline impl/client/src/main/javascript/web/pentaho/visual/views/echarts/Abstract.js:174]: No need to do.
My preference for code style is to test, return, test return. No elses.
For example:
```js
if(bgFillType === "none") {
  return "transparent";
}

if(bgFillType === "solid") {
  return bgColorFill;
}

var bgColorFill = model.backgroundColor;
// ...
return { ... };
```

**Review by dcleao — APPROVED** [review]

File: impl/client/src/main/javascript/web/pentaho/visual/views/echarts/Abstract.js (hunk 1 above)
Comment (AAravind27) [reply impl/client/src/main/javascript/web/pentaho/visual/views/echarts/Abstract.js]: Addressed.

**Comment by buildguy** [comment]
<div align='center'>

[👍 Frogbot scanned this pull request and found that it did not add vulnerable dependencies.](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>

<details>
<summary> <b>Note:</b> </summary>


---
<div align='center'>

**Frogbot** also supports **Contextual Analysis, Secret Detection, IaC and SAST Vulnerabilities Scanning**. This features are included as part of the [JFrog Advanced Security](https://jfrog.com/advanced-security) package, which isn't enabled on your system.

</div>


</details>


---
<div align='center'>

[🐸 JFrog Frogbot](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 9m 21s
Build command:
```console
mvn clean verify -B -e -Daudit -Djs.no.sandbox
```
---
:ok_hand: All tests passed!
 
Tests run: **5240**, Failures: **0**, Skipped: **5** &nbsp;&nbsp; [Test Results](https://wingman.orl.eng.hitachivantara.com/job/projects/job/common-ui/job/PR-1760/3/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Review by dcleao — APPROVED** [review]

**Pull request merged**